src/Utility/DgiUtils.php

```diff
@@ -66,6 +66,14 @@ public function executeEntityReactions(string $reaction_type, ContentEntityInter
    */
   public function getActiveReactionsForEntity(string $reaction_type, ContentEntityInterface $entity) {
     $this->provider->setEntity($entity);
+
+    // XXX: Need to force context re-evaluation to ensure that the entity being
+    // set by the provider in a single threaded scenario is evaluated once
+    // it has been changed.
+    $context_manager = new \ReflectionClass($this->contextManager);
+    $reset_context = $context_manager->getMethod('resetContextEvaluation');
+    $reset_context->setAccessible(TRUE);
+    $reset_context->invoke($this->contextManager);
```

Comment:
Just had the think, but this effectively creates a dependency on [Islandora's `context.manager` class replacement](https://github.com/Islandora/islandora/blob/3f42453cd478e572473b37c0634e022bec4c0b3c/src/IslandoraServiceProvider.php#L17-L18), as in, if there was other decoration of the service, or some other process that later replaced the class, then [the method](https://github.com/Islandora/islandora/blob/3f42453cd478e572473b37c0634e022bec4c0b3c/src/IslandoraContextManager.php#L16-L22) might not be available (due to a referencing a different class). Would throw a `ReflectionException`... unsure if we might want to catch/log/suppress, or... something else?

Reply:
Then again, further discussing w/ @jordandukart on Slack: `$context_manager` is type-hinted in the constructor as coming in as an `IslandoraContextManager`, so the dependency is already there.

Reply:
Chatted about this, given our desire to move away from `Context` in the long term and the current enforcement of the `IslandoraContextManager` in the constructor we are going to leave this as is.